recipes/opencv/4.x/conanfile.py

```diff
@@ -254,6 +278,12 @@ def _configure_cmake(self):
         if self.options.parallel:
             self._cmake.definitions["WITH_TBB"] = self.options.parallel == "tbb"
             self._cmake.definitions["WITH_OPENMP"] = self.options.parallel == "openmp"
+        if self.options.with_cuda:
+            self._cmake.definitions["WITH_CUDA"] = True
+            # This allows compilation on older GCC/NVCC, otherwise build errors.
+            self._cmake.definitions["CUDA_NVCC_FLAGS"] = "--expt-relaxed-constexpr"
+            if self.options.with_cublas:
+                self._cmake.definitions["WITH_CUBLAS"] = True
```

Comment:
```suggestion
        self._cmake.definitions["WITH_CUDA"] = if self.options.with_cuda
        # This allows compilation on older GCC/NVCC, otherwise build errors.
        if self.options.with_cuda:
            self._cmake.definitions["CUDA_NVCC_FLAGS"] = "--expt-relaxed-constexpr"
        self._cmake.definitions["WITH_CUBLAS"] = self.options.get_safe("with_cublas", False)
```

don't forget to remove those definitions at the beginning of `build()` to avoid duplicated definitions